src/app/features/city-staff-permissions/city-staff-permissions.component.spec.ts

```diff
@@ -40,15 +57,160 @@ describe('CityStaffPermissionsComponent', () => {
         MatDialogModule,
         RemoveMemberModule,
       ],
-      providers: [{ provide: HttpClient }],
-    }).compileComponents();
-
+    })
+      .overrideComponent(CityStaffPermissionsComponent, {
+        set: {
+          providers: [
+            {
+              provide: GetAllUsersService,
+              useClass: MockedGetAllUsersService,
+            },
+          ],
+        },
+      })
+      .compileComponents();
+  });
+  beforeEach(async () => {
     fixture = TestBed.createComponent(CityStaffPermissionsComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    _getAllUsersService = fixture.debugElement.injector.get(GetAllUsersService);
+    _matDialog = fixture.debugElement.injector.get(MatDialog);
+    dialogSpy = spyOn(TestBed.get(MatDialog), 'open').and.returnValue(
```

Comment:
Please look for another way to do this avoiding the use of the deprecated method.